src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,559 @@
+            return result;
+        }
+        for (String filename : configDropinXmls) {
+            result.addAll(getServerXmlFeatures(configDropinFolder, filename, null));
+        }
+        return result;
+    }
+
+    private Set<String> getServerXmlFeatures(File fileContext, String serverFile, List<String> parsedXmls) {
+        Set<String> result = new HashSet<String>();
+        File serverXml = new File(fileContext, serverFile);
+        List<String> updatedParsedXmls = new ArrayList<String>();
+        updatedParsedXmls.add(serverFile);
+        if (parsedXmls != null){
+            updatedParsedXmls.addAll(parsedXmls);
+        }
+        if (serverXml.exists()) {
+            try {
+                Document doc = new XmlDocument() {
+                    public Document getDocument(File file) throws IOException, ParserConfigurationException, SAXException {
+                        createDocument(file);
+                        return doc;
+                    }
+                }.getDocument(serverXml);
+                Element root = doc.getDocumentElement();
+                NodeList nodes = root.getChildNodes();
+                for (int i = 0; i < nodes.getLength(); i++) {
+                    if (nodes.item(i) instanceof Element) {
+                        Element child = (Element) nodes.item(i);
+                        if ("featureManager".equals(child.getNodeName())) {
+                            result.addAll(parseFeatureManagerNode(child));
+                        }
+                        if ("include".equals(child.getNodeName())){
+                            result = parseIncludeNode(result, child, fileContext, updatedParsedXmls);
+                        }
+                    }
+                }
+            } catch (IOException | ParserConfigurationException | SAXException e) {
+                warn("Failed to parse the xml file " + serverXml + ": " + e.getLocalizedMessage());
+                debug(e);
+            }
+        }
+        return result;
+    }
+    
+    private Set<String> parseIncludeNode(Set<String> origResult, Element node, File fileContext,
+            List<String> updatedParsedXmls) {
+        Set<String> result = origResult;
+        String includeFileName = node.getAttribute("location");
+        if (!updatedParsedXmls.contains(includeFileName)){
+            String onConflict = node.getAttribute("onConflict");
+            boolean onConflictMerge = Pattern.compile(Pattern.quote("MERGE"), Pattern.CASE_INSENSITIVE).matcher(onConflict).find();
+            boolean onConflictReplace = Pattern.compile(Pattern.quote("REPLACE"), Pattern.CASE_INSENSITIVE).matcher(onConflict).find();
+            boolean onConflictIgnore = Pattern.compile(Pattern.quote("IGNORE"), Pattern.CASE_INSENSITIVE).matcher(onConflict).find();
+            if (!(onConflictMerge || onConflictReplace || onConflictIgnore)){
+                onConflictMerge = true;
+            }
+            if (!onConflictIgnore){
```

Comment:
Where do you check in the code that a conflict actually occurs? Shouldn't you still add the features if there is no conflict?

Reply:
If ignore is specified, we verified that the entire file is ignored.

Reply:
@ericglau Does this happen even if there is no conflict? I would not expect that setting `onConflict="ignore"` is equivalent to not having the `include` at all.